packages/tocco-ui/src/Button/Button.js

```diff
@@ -1,92 +1,127 @@
 import PropTypes from 'prop-types'
 import React from 'react'
-import classNames from 'classnames'
+
+import Icon from '../Icon'
+import StyledButton from './StyledButton'
+import {
+  stylingAnimation,
+  stylingInk,
+  stylingLook,
+  stylingPosition
+} from '../utilStyles'
 
 /**
- * Styled Button
+ * Use <Button> to trigger any actions. Choose look and ink according Material Design.
  */
 const Button = props => {
-  const classes = classNames(
-    'btn',
-    props.className,
-    {
-      'btn-default': !props.primary,
-      'btn-primary': props.primary
-    }
-  )
-
-  const getIconClass = (icon, pending) => {
-    if (pending) {
-      return 'fa fa-circle-o-notch fa-spin'
-    }
-    return icon ? classNames({
-      'glyphicon': icon.startsWith('glyphicon-'),
-      'fa': icon.startsWith('fa-')
-    },
-    icon)
-      : ''
-  }
-
   return (
-    <button
+    <StyledButton
+      dense={props.dense}
+      disabled={props.disabled}
+      iconPosition={props.iconPosition}
+      ink={props.ink || props.buttonGroupInk || stylingInk.BASE}
+      look={props.look}
+      melt={props.buttonGroupMelt}
       name={props.name}
-      title={props.title}
       onClick={props.onClick}
-      className={classes}
-      disabled={props.disabled}
-      type={props.type ? props.type : 'button'}
-      style={props.style}
+      onMouseDown={props.onMouseDown}
+      onMouseEnter={props.onMouseEnter}
+      onMouseLeave={props.onMouseLeave}
+      title={props.title}
+      type={props.type}
     >
-      <i className={getIconClass(props.icon, props.pending)}/> {props.label || ''}
-    </button>)
+      {props.icon && <Icon
+        dense={props.dense}
+        icon={props.icon}
+        position={props.label ? props.iconPosition : stylingPosition.solely}/>}
+      {props.pending && <Icon
+        animation={stylingAnimation.SPIN}
+        dense={props.dense}
+        icon="fa-circle-o-notch"
+        position={props.label ? props.iconPosition : stylingPosition.solely}/>}
+      {props.label && <span>{props.label}</span>}
+    </StyledButton>
+  )
+}
+
+Button.defaultProps = {
+  iconPosition: stylingPosition.BEFORE,
+  look: stylingLook.FLAT,
+  type: 'button'
 }
 
 Button.propTypes = {
   /**
-   * Will be displayed on button, default is an empty string
+   * May be passed from <ButtonGroup> to use as default for ink. Do not set manually.
    */
-  label: PropTypes.node,
+  buttonGroupInk: PropTypes.oneOf([stylingInk.BASE, stylingInk.PRIMARY]),
```

Comment:
Striving for DRY-ness is of course important. In this case I am not convenienced yet, since it decreases readability. Searching for "PropTypes.oneOf" reveals 49 matches, hence there is a real potential for DRY-ness, still exist many variants. We might introduce only the variants oneOfAnimation, oneOfInk, oneOfLook and oneOfPosition in stylingConstants.js.

Reply:
yes that what i said more or less?

I just think inkPropType is the better fitting name than oneOfInk

Reply:
see 9799f40a5fafad4181eac013f17902e63b82c0d5